static/usage/item/helper-error/react.md

```diff
@@ -0,0 +1,41 @@
+```tsx
+import React, { useState } from 'react';
+import { IonInput, IonItem, IonLabel, IonNote } from '@ionic/react';
+
+function Example() {
+  const [isTouched, setIsTouched] = useState(false);
+  const [isValid, setIsValid] = useState(false);
+  const [isInvalid, setIsInvalid] = useState(false);
+
+  const validateEmail = (email: string) => {
+    return email.match(/^(?=.{1,254}$)(?=.{1,64}@)[a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+)*@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/);
+  }
+
+  const validate = (ev: Event) => {
+    const value = (ev.target as HTMLInputElement).value;
+
+    setIsValid(false);
+    setIsInvalid(false);
+
+    if (value === '') return;
+
+    validateEmail(value) !== null ? setIsValid(true) : setIsInvalid(true);
+  }
+
+  const markTouched = () => {
+    setIsTouched(true);
+  }
+
+  return (
+    <>
+      <IonItem fill="solid" className={`${isValid && "ion-valid"} ${isInvalid && "ion-invalid"} ${isTouched && "ion-touched"}`}>
+        <IonLabel position="floating">Email</IonLabel>
+        <IonInput type="email" onIonInput={(event) => validate(event)} onIonBlur={() => markTouched()}></IonInput>
+        <IonNote slot="helper">Enter a valid email</IonNote>
+        <IonNote slot="error">Invalid email</IonNote>
```

Comment:
```suggestion
      <IonLabel position="floating">Email</IonLabel>
      <IonInput type="email" onIonInput={(event) => validate(event)} onIonBlur={() => markTouched()}></IonInput>
      <IonNote slot="helper">Enter a valid email</IonNote>
      <IonNote slot="error">Invalid email</IonNote>
```

Reply:
This indentation is needed since these are all children of the `IonItem`

Reply:
Whoops, my bad, misread the HTML 😆 In that case you can remove the `<>`/`</>` here too.